test/e2e/serial/workload_placement_test.go

```diff
@@ -715,6 +717,305 @@ var _ = Describe("[serial][disruptive][scheduler] workload placement", func() {
+			// so in order to be certain that the pod will land on different node we need to request more than 7 CPUs in total
+			requiredRes = corev1.ResourceList{
+				// 6 here + 2 on the second container
+				corev1.ResourceCPU:    resource.MustParse("6"),
+				corev1.ResourceMemory: resource.MustParse("100Mi"),
+			}
+
+			err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), updatedDp)
+			Expect(err).ToNot(HaveOccurred())
+
+			podSpec = &updatedDp.Spec.Template.Spec
+			podSpec.Containers[0].Resources.Requests = requiredRes
+			podSpec.Containers[0].Resources.Limits = requiredRes
+
+			err = fxt.Client.Update(context.TODO(), updatedDp)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = e2ewait.ForDeploymentComplete(fxt.Client, dp, time.Second*10, time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = fxt.Client.Get(context.TODO(), client.ObjectKeyFromObject(dp), updatedDp)
+			Expect(err).ToNot(HaveOccurred())
+
+			namespacedDpName = fmt.Sprintf("%s/%s", updatedDp.Namespace, updatedDp.Name)
+			Eventually(func() bool {
+				pods, err = schedutils.ListPodsByDeployment(fxt.Client, *updatedDp)
+				if err != nil {
+					klog.Warningf("failed to list the pods of deployment: %q error: %v", namespacedDpName, err)
+					return false
+				}
+				if len(pods) != 1 {
+					klog.Warningf("%d pods are exists under deployment %q", len(pods), namespacedDpName)
+					return false
+				}
+				return true
+			}, time.Minute, 5*time.Second).Should(BeTrue(), "there should be only one pod under deployment: %q", namespacedDpName)
+
+			nrtPostUpdateNo2DeploymentList, err := e2enrt.GetUpdated(fxt.Client, nrtPostCreateDeploymentList, time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedPod = pods[0]
+			By(fmt.Sprintf("checking the pod landed on a node which is different than target node %q vs %q", targetNodeName, updatedPod.Spec.NodeName))
+			Expect(updatedPod.Spec.NodeName).ToNot(Equal(targetNodeName),
+				"pod should not landed on node %q", targetNodeName)
+
+			By(fmt.Sprintf("checking the pod was scheduled with the topology aware scheduler %q", schedulerName))
+			schedOK, err = nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod.Namespace, updatedPod.Name, schedulerName)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod.Namespace, updatedPod.Name, schedulerName)
+
+			rl = e2enrt.ResourcesFromGuaranteedPod(updatedPod)
+
+			nrtPostUpdateNo2, err := e2enrt.FindFromList(nrtPostUpdateNo2DeploymentList.Items, updatedPod.Spec.NodeName)
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("checking NRT for target node %q updated correctly", targetNodeName))
+			// TODO: this is only partially correct. We should check with NUMA zone granularity (not with NODE granularity)
+			_, err = e2enrt.CheckZoneConsumedResourcesAtLeast(*nrtPostUpdate, *nrtPostUpdateNo2, rl)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = unlabel()
```

Comment:
let's make sure we unlabel inconditionally - e.g. even if the tests fail with error, so we don't leave leftovers around

Reply:
sure I'll pass it to the ```AfterEach()```

Reply:
Adding it on `AfterEach` to remove any labels we may have left over seems like a good approach. Works for me.